File: VoidSchach960/src/test/java/figures/QueenTest.java
package figures;

import java.util.LinkedList;
import java.util.List;

import junit.framework.TestCase;
import image.*;
import board.*;
import helper.*;

import static org.mockito.Mockito.mock;
/**
 * @author stephan
 */
public class QueenTest extends TestCase
{
	public QueenTest(String arg0)
	{
		super(arg0);
	}
	
	public void testIsReachable()
	{
		String des     = "white 0 Queen-white-e2";
		ChessGame game = new ChessGame( des );
		
		FigureImage figureImage = new FigureImageMock(10,20,30);
		Position from           = Position.get( "e2" );
		Position to1            = Position.get( "e5" );
		Position to2            = Position.get( "c4" );
		Position to3            = Position.get( "d4" );
    
		Queen queen = new Queen(figureImage,true,from);
		assertTrue( queen.isReachable(to1,game) );
		assertTrue( queen.isReachable(to2,game) );
		assertFalse( queen.isReachable(to3,game) );
		assertFalse( queen.isReachable(from,game) );
	}
	
	public void testGetPossibleMoves()
	{
		String des     = "white 0 Queen-white-d1 King-white-e1-0 ";
		SimpleArrayBoard game = new SimpleArrayBoard( des, mock(LastMoveProvider.class) );
    
		Figure queen = game.getFigure( Position.get( "d1" ) );
		List<Move> moveIter = new LinkedList<>();
		queen.getPossibleMoves(game,moveIter);
		assertEquals( moveIter.size(),17 );
	}

}

File: module-ui/src/test/kotlin/voidchess/figures/QueenTest.kt
package voidchess.figures

import org.testng.annotations.DataProvider
import org.testng.annotations.Test
import voidchess.board.ChessGame
import voidchess.board.ArrayChessBoard
import voidchess.board.getFigure
import voidchess.common.board.move.Position
import voidchess.figures.Queen

import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue


class QueenTest {
    @Test
    fun testIsReachable() {
        val des = "white 0 Queen-white-e2  King-white-e1-0  King-black-e8-0"
        val game = ChessGame(des)

        val from = Position.byCode("e2")
        val to1 = Position.byCode("e5")
        val to2 = Position.byCode("c4")
        val to3 = Position.byCode("d4")
        val to4 = Position.byCode("e1")
        val to5 = Position.byCode("e8")

        val queen = Queen(true, from)
        assertTrue(queen.isReachable(to1, game))
        assertTrue(queen.isReachable(to2, game))
        assertFalse(queen.isReachable(to3, game))
        assertFalse(queen.isReachable(to4, game))
        assertTrue(queen.isReachable(to5, game))
        assertFalse(queen.isReachable(from, game))
    }

    @Test(dataProvider = "getTestIsSelectableData")
    fun testIsSelectable(des: String, figurePos: String, expectedIsSelectable: Boolean) {
        val game = ArrayChessBoard(des)
        val queen = game.getFigure(Position.byCode(figurePos))
        assertEquals(expectedIsSelectable, queen.isSelectable(game), "isSelectable")
    }

    @DataProvider
    fun getTestIsSelectableData(): Array<Array<Any>> = arrayOf(
            arrayOf("white 0 King-white-e1-0 Bishop-black-c3 Queen-white-b4 King-black-e8-0", "b4", true),
            arrayOf("white 0 King-white-e1-0 Bishop-black-c3 Queen-white-b5 King-black-f8-1", "b5", false)
    )
}
